src/main/java/mServer/crawler/sender/MediathekZdfTivi.java

```diff
@@ -317,7 +301,39 @@ private void addTivi_(String url) {
                 Log.errorLog(454123698, ex);
             }
         }
+        
+        private long parseDauer(MSStringBuilder seite, String url) {
+            long dauer = 0;
+        
+            try {
+                String duration = seite.extract("<duration>", "<");
+                if (!duration.isEmpty()) {
+                    dauer = parseDuration(duration);
+                }
+                if (dauer == 0) {
+                    String length = seite.extract("<lengthSec>", "<");
+                    dauer = Integer.parseInt(length);
+                }
+            } catch (NumberFormatException ex) {
+                Log.errorLog(349761012, ex, "Dauer: " + url);
+            }
 
+            return dauer;
+        }
+        
+        // Parst Längenangaben im Format P0Y0M0DT0H22M6.000S
+        private long parseDuration(String duration) throws NumberFormatException {
+            
+            duration = duration.replace("P0Y0M0DT", "");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "duration". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1226)